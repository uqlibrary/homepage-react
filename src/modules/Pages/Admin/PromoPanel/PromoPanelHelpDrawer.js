import React from 'react';
import PropTypes from 'prop-types';

import SimpleBackdrop from '@mui/material/Backdrop';
import Button from '@mui/material/Button';
import Fade from '@mui/material/Fade';
import { makeStyles } from '@mui/styles';
import Drawer from '@mui/material/Drawer';

const useStyles = makeStyles(
    theme => ({
        drawer: {
            overflowY: 'scroll',
            '& p': {
                marginBlockStart: 0,
                marginBlockEnd: '1em',
            },
            '& li': {
                marginBlockStart: 0,
                marginBlockEnd: '1em',
            },
            '& dt': {
                fontStyle: 'italic',
            },
        },
        paper: {
            backgroundColor: theme.palette.background.paper,
            padding: theme.spacing(2, 4, 3),
            width: 500,
            [theme.breakpoints.down('md')]: {
                width: 200,
            },
        },
    }),
    { withTheme: true },
);
export const PromoPanelHelpDrawer = ({ open, helpContent, closeHelpLightbox }) => {
    const classes = useStyles();

    return (
        <Drawer
            anchor="right"
            className={classes.drawer}
            open={open}
            onClose={closeHelpLightbox}
            closeAfterTransition
            BackdropComponent={SimpleBackdrop}
            BackdropProps={{
                timeout: 500,
            }}
        >
            <Fade in={open}>
                <div className={classes.paper}>
                    <h2 data-testid="help-drawer-title" id="help-drawer-title">
                        {helpContent?.title || /* istanbul ignore next */ 'TBA'}
                    </h2>
                    <div>{helpContent?.text || /* istanbul ignore next */ ''}</div>
                    <div>
                        <Button
                            variant="contained"
                            color="primary"
                            className={classes.button}
                            children={helpContent?.buttonLabel || 'Close'}
                            data-testid="promopanel-helpdrawer-close-button"
                            id="promopanel-helpdrawer-close-button"
                            onClick={closeHelpLightbox}
                        />
                    </div>
                </div>
            </Fade>
        </Drawer>
    );
};

PromoPanelHelpDrawer.propTypes = {
    helpContent: PropTypes.any,
    open: PropTypes.any,
    setHelpLightboxOpen: PropTypes.any,
    closeHelpLightbox: PropTypes.any,
};

export default PromoPanelHelpDrawer;
