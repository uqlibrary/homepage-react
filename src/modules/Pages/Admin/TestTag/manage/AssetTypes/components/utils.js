export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, rows: {}, row: {}, title: '' };

export const actionReducer = (_, action) => {
    switch (action.type) {
        case 'add':
            return {
                title: 'Add Asset Type',
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: { asset_type_id: 'auto' },
            };
        case 'edit':
            return { title: 'Edit Asset Type', isAdd: false, isEdit: true, isDelete: false, row: action.row };
        case 'clear':
            return { ...emptyActionState };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, row: action.row };
        default:
            throw `Unknown action '${action.type}'`;
    }
};
