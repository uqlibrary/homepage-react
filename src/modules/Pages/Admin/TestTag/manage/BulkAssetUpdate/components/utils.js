import { createLocationString } from '../../../helpers/helpers';

export const transformRow = row => {
    return row.map(line => {
        if (!!line?.asset_location) return line;
        return {
            ...line,
            asset_type_name: line?.asset_type?.asset_type_name ?? '',
            asset_location: !!line?.last_location
                ? createLocationString({
                      site: line.last_location.site_name,
                      building: line.last_location.building_name,
                      floor: line.last_location.floor_id_displayed,
                      room: line.last_location.room_id_displayed,
                  })
                : '',
        };
    });
};

export const transformRequest = formValues => {
    if (!!formValues.hasDiscardStatus) {
        formValues.hasLocation = false;
        formValues.hasAssetType = false;
    } else {
        formValues.hasDiscardStatus = false;
    }
    return {
        asset: formValues.asset_list.reduce((cumulative, current) => [...cumulative, current.asset_id], []),
        ...(!!formValues.hasLocation ? { asset_room_id_last_seen: formValues.location.room } : {}),
        ...(!!formValues.hasAssetType ? { asset_type_id: formValues.asset_type.asset_type_id } : {}),
        ...(!!formValues.hasDiscardStatus ? { is_discarding: 1, discard_reason: formValues.discard_reason } : {}),
    };
};

export const transformFilterRow = row => {
    return row.map(line => {
        if (!!line?.asset_id_displayed) return line;
        return {
            ...line,
            asset_id_displayed: line?.asset_barcode ?? '',
            asset_location: createLocationString({
                site: line.site_name,
                building: line.building_name,
                floor: line.floor_id_displayed,
                room: line.room_id_displayed,
            }),
        };
    });
};
