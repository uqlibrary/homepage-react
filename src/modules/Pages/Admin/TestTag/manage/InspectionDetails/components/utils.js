import moment from 'moment';
import locale from '../../../testTag.locale';
const dateFormat = locale.config.format.dateFormatNoTime;

export const transformRow = row => {
    return row.map(line => {
        if (!!row.user_name) return row;
        return {
            asset_id: line.asset_id,
            asset_id_displayed: line.asset_id_displayed,
            asset_type: line.asset_type?.asset_type_name,
            asset_status: line.asset_status,
            user_name: line.last_inspection?.user_name,
            inspect_date: !!line.last_inspection ? moment(line.last_inspection.inspect_date).format(dateFormat) : '',
            inspect_notes: line.last_inspection?.inspect_notes ?? '',
            inspect_fail_reason: line.last_inspection?.inspect_fail_reason ?? '',
            discard_reason: line.last_discard?.discard_reason ?? '',
            repairer_name: line.last_repair?.repairer_name ?? '',
            last_inspect_status: line.last_inspection?.inspect_status ?? '',
        };
    });
};

export const transformUpdateRequest = request => {
    const clonedRequest = structuredClone(request);
    const { asset_status: assetStatus, last_inspect_status: lastInspectStatus } = clonedRequest;
    const defaultValues = { inspect_notes: clonedRequest.inspect_notes };

    switch (assetStatus) {
        case locale.config.assetStatus.current:
            return defaultValues;
        case locale.config.assetStatus.failed:
            return { ...defaultValues, inspect_fail_reason: clonedRequest.inspect_fail_reason };
        case locale.config.assetStatus.outforrepair:
            return {
                ...defaultValues,
                inspect_fail_reason: clonedRequest.inspect_fail_reason,
                repairer_name: clonedRequest.repairer_name,
            };
        case locale.config.assetStatus.discarded:
            return {
                ...defaultValues,
                ...(lastInspectStatus === locale.config.inspectStatus.failed
                    ? { inspect_fail_reason: clonedRequest.inspect_fail_reason }
                    : {}),
                discard_reason: clonedRequest.discard_reason,
            };
        default:
            return request;
    }
};
