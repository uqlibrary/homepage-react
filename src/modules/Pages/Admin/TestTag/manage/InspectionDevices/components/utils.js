import moment from 'moment';
import locale from 'modules/Pages/Admin/TestTag/testTag.locale';
const dateFormat = locale.pages.manage.config.dateFormat;

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, title: '', row: {} };
export const actionReducer = (_, action) => {
    const { type, row, title, ...props } = action;
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: {
                    device_id: 'auto',
                    device_calibrated_date_last: moment().format(dateFormat),
                    device_calibration_due_date: moment()
                        .add(1, 'd')
                        .format(dateFormat),
                },
                title,
                props: { ...props },
            };
        case 'edit':
            return { isAdd: false, isEdit: true, isDelete: false, title, row, props: { ...props } };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, title, row, props: { ...props } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};

export const formatDateStrings = (row, suffix) => {
    const dateLastParts = row.device_calibrated_date_last.split(' ');
    const dateNextParts = row.device_calibration_due_date.split(' ');
    return {
        ...row,
        device_calibrated_date_last:
            dateLastParts.length > 1 ? row.device_calibrated_date_last : `${row.device_calibrated_date_last} ${suffix}`,
        device_calibration_due_date:
            dateNextParts.length > 1 ? row.device_calibration_due_date : `${row.device_calibration_due_date} ${suffix}`,
    };
};

export const transformAddRequest = (request, user) => {
    delete request.device_id;
    delete request.device_current_flag;

    const newResponse = { ...formatDateStrings(request, '00:00:00'), device_department: user.user_department };
    return newResponse;
};

export const transformUpdateRequest = request => {
    delete request.device_current_flag;

    const newResponse = formatDateStrings(request, '00:00:00');
    return newResponse;
};

export const transformRow = row => {
    return row.map(line => ({
        ...line,
        device_calibrated_date_last: moment(line.device_calibrated_date_last).format(
            locale.config.format.dateFormatNoTime,
        ),
        device_calibration_due_date: moment(line.device_calibration_due_date).format(
            locale.config.format.dateFormatNoTime,
        ),
    }));
};
