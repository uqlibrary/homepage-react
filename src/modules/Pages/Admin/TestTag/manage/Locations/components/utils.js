import React from 'react';
import RowMenuCell from './../../../SharedComponents/DataTable/RowMenuCell';

export const createLocationString = ({ site, building, floor }) =>
    [site, building, floor].filter(item => !!item).join(' / ');

export const getColumns = ({ config, locale, selectedFilter, handleEditClick, handleDeleteClick }) => {
    const actionsCell = {
        field: 'actions',
        headerName: locale?.actions,
        renderCell: params => {
            return (
                <RowMenuCell
                    {...params}
                    handleEditClick={handleEditClick}
                    {...((params.row?.asset_count ?? 1) === 0 ? { handleDeleteClick: handleDeleteClick } : {})}
                />
            );
        },
        sortable: false,
        width: 100,
        headerAlign: 'center',
        filterable: false,
        align: 'center',
        disableColumnMenu: true,
        disableReorder: true,
        renderInUpdate: false,
    };

    const columns = [];
    const keys = Object.keys(config[selectedFilter].fields);

    keys.forEach(key => {
        !!(config[selectedFilter].fields[key]?.fieldParams.renderInTable ?? true) &&
            columns.push({
                field: key,
                headerName: locale?.[selectedFilter]?.[key].label,
                editable: false,
                sortable: false,
                ...config[selectedFilter].fields[key].fieldParams,
            });
    });

    columns && columns.length > 0 && columns.push(actionsCell);
    return columns;
};

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, row: {} };
export const actionReducer = (_, action) => {
    const { type, row, selectedFilter, ...props } = action;
    console.log('actionReducer', type, props);
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: { [`${selectedFilter}_id`]: 'auto' },
                props: { ...props },
            };
        case 'edit':
            return { isAdd: false, isEdit: true, isDelete: false, row, props: { ...props } };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, row, props: { ...props, selectedFilter } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};

export const transformAddRequest = ({ request, selectedFilter, location }) => {
    // add requests may have an id field (probably 'auto') that needs to be removed
    delete request[`${selectedFilter}_id`];

    // get location key names i.e. ['site','building','floor','room']
    const keys = Object.keys(location);

    // get current selected location index
    const index = keys.findIndex(key => key === selectedFilter);

    // index 0 (sites) doesnt have a preceding location, so just return the request
    if (index === 0) return request;

    // else grab the name of the previous location type i.e. if selectedFilter is 'room', this would return 'building'
    const prevKey = keys[index - 1];

    // build a new request by inserting a key:value in format e.g. 'room_floor_id: 1'
    return { ...request, [`${selectedFilter}_${prevKey}_id`]: location[prevKey] };
};

export const transformUpdateRequest = ({ request, selectedFilter, location }) => {
    // get location key names i.e. ['site','building','floor','room']
    const keys = Object.keys(location);

    // get current selected location index
    const index = keys.findIndex(key => key === selectedFilter);

    // index max (room) doesnt have any superseding locations, so just return the request
    if (index === keys.length - 1) return request;

    // else grab the name of the next location type i.e. if selectedFilter is 'building', this would return 'room'
    const nextKey = keys[index + 1];

    // now remove the array of next location data we got from the server e.g.
    // for sites, we also get a buildings:[] array (note plural), for buildings
    // we also get a rooms:[] array etc.
    // We dont need to send all of this stuff back to the server.
    delete request[`${nextKey}s`];

    return request;
};
