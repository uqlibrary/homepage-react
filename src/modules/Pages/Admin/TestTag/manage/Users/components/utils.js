export const transformRow = row => {
    return row.map(line => ({
        ...line,
        can_admin_cb: line?.privileges?.can_admin === 1,
        can_inspect_cb: line?.privileges?.can_inspect === 1,
        can_alter_cb: line?.privileges?.can_alter === 1,
        can_see_reports_cb: line?.privileges?.can_see_reports === 1,
        user_current_flag_cb: line?.user_current_flag === 1,
        can_admin: line?.privileges?.can_admin === 1 ? 'Yes' : 'No',
        can_inspect: line?.privileges?.can_inspect === 1 ? 'Yes' : 'No',
        can_alter: line?.privileges?.can_alter === 1 ? 'Yes' : 'No',
        can_see_reports: line?.privileges?.can_see_reports === 1 ? 'Yes' : 'No',
        user_current_flag: line?.user_current_flag === 1 ? 'Yes' : 'No',
    }));
};

export const emptyActionState = { isAdd: false, isEdit: false, isDelete: false, title: '', row: {} };

export const transformUpdateRequest = request => {
    // delete request.previleges;
    delete request.inspectionCount;
    delete request.id;
    request.privileges.can_admin = request?.can_admin_cb ? 1 : 0;
    request.privileges.can_inspect = request?.can_inspect_cb ? 1 : 0;
    request.privileges.can_alter = request?.can_alter_cb ? 1 : 0;
    request.privileges.can_see_reports = request?.can_see_reports_cb ? 1 : 0;

    request.user_current_flag = request?.user_current_flag_cb ? 1 : 0;

    delete request.can_admin;
    delete request.can_admin_cb;
    delete request.can_inspect;
    delete request.can_inspect_cb;
    delete request.can_alter;
    delete request.can_alter_cb;
    delete request.can_see_reports;
    delete request.can_see_reports_cb;

    return request;
};

export const transformAddRequest = request => {
    // delete request.previleges;
    console.log('REquest is ', request);
    delete request.inspectionCount;
    delete request.id;

    request.privileges.can_admin = request?.can_admin_cb ? 1 : 0;
    request.privileges.can_inspect = request?.can_inspect_cb ? 1 : 0;
    request.privileges.can_alter = request?.can_alter_cb ? 1 : 0;
    request.privileges.can_see_reports = request?.can_see_reports_cb ? 1 : 0;

    request.user_current_flag = request?.user_current_flag_cb ? 1 : 0;

    delete request.can_admin;
    delete request.can_admin_cb;
    delete request.can_inspect;
    delete request.can_inspect_cb;
    delete request.can_alter;
    delete request.can_alter_cb;
    delete request.can_see_reports;
    delete request.can_see_reports_cb;

    return request;
};

export const actionReducer = (_, action) => {
    const { type, row, title, ...props } = action;
    switch (type) {
        case 'add':
            return {
                isAdd: true,
                isEdit: false,
                isDelete: false,
                row: {
                    user_id: 'Auto',
                    user_uid: '',
                    user_name: '',
                    user_current_flag_cb: true,
                },
                title,
                props: { ...props },
            };
        case 'edit':
            console.log('Edit Row', row);
            return {
                isAdd: false,
                isEdit: true,
                isDelete: false,
                title,
                row: { ...row, id: row.user_id },
                // row,
                props: { ...props },
            };
        case 'delete':
            return { isAdd: false, isEdit: false, isDelete: true, title, row, props: { ...props } };
        case 'clear':
            return { ...emptyActionState };
        default:
            throw `Unknown action '${type}'`;
    }
};
