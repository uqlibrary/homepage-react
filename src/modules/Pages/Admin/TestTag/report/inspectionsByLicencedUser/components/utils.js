import locale from '../../../testTag.locale';
const moment = require('moment');

export const getNameStyles = (name, inspectorName, theme) => ({
    fontWeight:
        inspectorName.indexOf(name) === -1 ? theme.typography.fontWeightRegular : theme.typography.fontWeightMedium,
});

export const transformRow = row => {
    return row.map(line => ({
        ...line,
        start_date: !!line?.start_date ? moment(line.start_date).format(locale.config.format.dateFormatNoTime) : '--',
        end_date: !!line?.start_date ? moment(line.end_date).format(locale.config.format.dateFormatNoTime) : '--',
    }));
};
