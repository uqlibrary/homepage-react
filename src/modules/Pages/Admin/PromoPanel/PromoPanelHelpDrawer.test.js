import { PromoPanelHelpDrawer } from './PromoPanelHelpDrawer';
import React from 'react';
import { rtlRender, fireEvent } from '../../../../../utils/test-utils';
import { default as locale } from 'modules/Pages/Admin/PromoPanel/promopanel.locale';

function setup(testProps = {}) {
    const props = {
        helpContent: locale.listPage.help,
        open: true,
        closeHelpLightbox: jest.fn(),
        ...testProps,
    };
    return rtlRender(<PromoPanelHelpDrawer {...props} {...testProps} />);
}

describe('Component Help Drawer', () => {
    it('should render correctly', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });
    it('should close correctly', () => {
        const closeFn = jest.fn();
        const { getByTestId } = setup({ closeHelpLightbox: closeFn });
        fireEvent.click(getByTestId('promopanel-helpdrawer-close-button'));
        expect(closeFn).toHaveBeenCalledTimes(1);
    });
});
