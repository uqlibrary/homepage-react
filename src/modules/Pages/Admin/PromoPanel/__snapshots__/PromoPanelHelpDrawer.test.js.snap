// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Help Drawer should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal makeStyles-drawer-1"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiPaper-elevation16"
        tabindex="-1"
      >
        <div
          class="makeStyles-paper-2"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <h2
            data-testid="help-drawer-title"
            id="help-drawer-title"
          >
            Promo Panels
          </h2>
          <div>
            <div>
              <p>
                The Promotional Panel Management Application enables you (an authorised user) to set promotional messages for library users via the Promo Panel Box on the website homepage.
              </p>
              <p>
                You can:
              </p>
              <ul>
                <li>
                  add, update and remove scheduled messages
                </li>
                <li>
                  target the logged-out homepage and select logged-in user groups e.g. UQ students, HDR students or UQ staff
                </li>
                <li>
                  set default (fallback) messages for user groups for times when no scheduled messages are set.
                </li>
              </ul>
              <h3>
                How to manage promo panels
              </h3>
              <h4>
                Preview (view) a panel
              </h4>
              <p>
                Press the 
                <strong>
                  View 
                </strong>
                button beside a panel. A preview window will appear showing the style promo box (title and message).
              </p>
              <p>
                Click off the Preview window to close it.
              </p>
              <h4>
                Add a panel
              </h4>
              <p>
                To reuse an existing message, use 
                <strong>
                  Clone
                </strong>
                 - 
                <strong>
                  More actions (downward arrow) 
                </strong>
                button &gt; 
                <strong>
                  Clone 
                </strong>
                button - beside an existing panel. 
              </p>
              <p>
                To create a new promo panel press the 
                <strong>
                  Add panel 
                </strong>
                button. The Create a promo form will appear.
              </p>
              <p>
                Scheduled promos will appear on the website within two minutes of their start time. Hard refresh your browser to view the promo panel.
              </p>
              <h4>
                Edit a panel
              </h4>
              <p>
                Press 
                <strong>
                  More actions (downward arrow) 
                </strong>
                &gt; 
                <strong>
                  Edit 
                </strong>
                beside a panel on the listing screen. The Edit a promo form will appear with fields prefilled.
              </p>
              <p>
                <strong>
                  Check the user groups for a promo before changing the content 
                </strong>
                as a panel may affect multiple user groups. 
              </p>
              <p>
                Updated panels will change on the website within two minutes. Hard refresh your browser to view the updated message.
              </p>
              <h4>
                Clone (copy) a panel
              </h4>
              <p>
                Press the 
                <strong>
                  More actions (downward arrow) 
                </strong>
                &gt; 
                <strong>
                  Clone
                </strong>
                 button beside a panel. The Clone a promo form will appear with the Admin notes, Title and Message fields prefilled
                <em>
                  .
                </em>
              </p>
              <p>
                Your cloned panel will appear on the website within two minutes of its scheduled start time. Hard refresh your browser to view the updated message.
              </p>
              <h4>
                Unschedule a panel
              </h4>
              <p>
                <strong>
                  Default and scheduled panels table only.
                </strong>
              </p>
              <p>
                To expire a message that you would like to end early, 
                <strong>
                  Edit 
                </strong>
                the panel and adjust its end date. The panel will expire and move to the Past panels table.
              </p>
              <p>
                To unschedule a message that hasn't run to be used at some unknown future time, press 
                <strong>
                  More actions (downward arrow) 
                </strong>
                &gt; 
                <strong>
                  Unschedule
                </strong>
                .
              </p>
              <p>
                The panel (admin notes, title and message) will be moved to the Unallocated panels table to be used later or deleted.
              </p>
              <p>
                <strong>
                  Delete a panel
                </strong>
              </p>
              <p>
                <strong>
                  Unallocated panels table only
                </strong>
                .
              </p>
              <p>
                Delete should generally only be used for messages that we did not end up displaying to clients so we can keep track of our promos.
              </p>
              <p>
                If a scheduled promo ran for clients for any period of time, it should be expired by changing the scheduled end date and moving the promo to the Past promos table.
              </p>
              <p>
                To delete a promo from the Unallocated panels table:
              </p>
              <ol>
                <li>
                  Press the 
                  <strong>
                    More actions (downward arrow) 
                  </strong>
                  &gt; 
                  <strong>
                    Delete 
                  </strong>
                  button
                  <br />
                  <br />
                </li>
                <li>
                  Confirm you wish to proceed. Deleted panels cannot be restored.
                </li>
              </ol>
              <h3>
                About the Promo Panel listing screen
              </h3>
              <p>
                The Listing Screen has several tables:
              </p>
              <ul>
                <li>
                  <strong>
                    Currently shown panels - 
                  </strong>
                  a summary of the promo running now for each user group and whether it is a default or scheduled message.
                  <br />
                  <br />
                </li>
                <li>
                  <strong>
                    Default and scheduled panels 
                  </strong>
                  <strong>
                    - 
                  </strong>
                  lists the scheduled messages and the default promos for each user group based on the start (From) date. The default user group promo is listed last. The table can be filtered by user group.
                  <br />
                  <br />
                </li>
                <li>
                  <strong>
                    Unallocated panels - 
                  </strong>
                  lists panels (i.e. at least as title and message) without a schedule (a start and end date/time). These promos can be scheduled, be set as a default message, or deleted.
                  <br />
                  <br />
                </li>
                <li>
                  <strong>
                    Past panels - 
                  </strong>
                  lists past scheduled promotions (does not include defaults yet). View or clone past messages you wish to use again.
                </li>
              </ul>
              <h4>
                Schedule panel and set default buttons
              </h4>
              <p>
                Use the 
                <strong>
                  Schedule panel
                </strong>
                 and
                <strong>
                   Set 
                </strong>
                default buttons for a particular user group in the Default and Scheduled panels table to choose a promo from the Unallocated panels table and:
              </p>
              <ul>
                <li>
                  <strong>
                    Schedule panel - 
                  </strong>
                  provide a schedule (start and end date)
                </li>
                <li>
                  <strong>
                    Set default 
                  </strong>
                  - set it as the default message
                </li>
              </ul>
              <p>
                for this user group only.
              </p>
              <h4>
                Promo action buttons
              </h4>
              <p>
                Each promotion has two action buttons. There is a primary action and a 
                <strong>
                  More actions (
                </strong>
                <strong>
                  downwards arrow) 
                </strong>
                button. Use the actions to interact with the messages: 
              </p>
              <ul>
                <li>
                  <strong>
                    Default and scheduled
                  </strong>
                  : View (primary), Edit, Clone and Unschedule
                  <br />
                  <br />
                </li>
                <li>
                  <strong>
                    Unallocated
                  </strong>
                  : View (primary), Edit, Clone and Delete
                  <br />
                  <br />
                </li>
                <li>
                  <strong>
                    Past: 
                  </strong>
                  View (primary) and clone.
                </li>
              </ul>
            </div>
          </div>
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="promopanel-helpdrawer-close-button"
              id="promopanel-helpdrawer-close-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Close
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
